htdocs/tests/Unit/class/libraries/Xmf/Database/TablesTest.php

```diff
@@ -0,0 +1,383 @@
+     */
+    protected $object;
+    /**
+     * @var string
+     */
+    protected $prefix;
+
+    /**
+     * Sets up the fixture, for example, opens a network connection.
+     * This method is called before a test is executed.
+     */
+    protected function setUp(): void
+    {
+        $this->object = new Tables();
+        $this->prefix = \XoopsDatabaseFactory::getDatabaseConnection()->prefix();
+    }
+
+    /**
+     * Tears down the fixture, for example, closes a network connection.
+     * This method is called after a test is executed.
+     */
+    protected function tearDown(): void
+    {
+    }
+
+    protected function prefix($table)
+    {
+        return $this->prefix . '_' . $table;
+    }
+
+    public function testName()
+    {
+        $tableName = 'users';
+        $actual    = $this->object->useTable($tableName);
+        $this->assertTrue($actual);
+
+        $tables = $this->object->dumpTables();
+        $this->assertArrayHasKey($tableName, $tables);
+        $this->assertEquals($this->prefix($tableName), $tables[$tableName]['name']);
+    }
+
+    public function testAddColumn()
+    {
+        $tableName = 'users';
+        $actual    = $this->object->useTable($tableName);
+        $this->assertTrue($actual);
+
+        $columnName = 'not_real_column';
+        $columnAttr = 'int NOT NULL';
+
+        $this->object->addColumn($tableName, $columnName, $columnAttr);
+        $queue    = $this->object->dumpQueue();
+        $expected = "ALTER TABLE `{$this->prefix}_{$tableName}` ADD COLUMN `{$columnName}` {$columnAttr}";
+        $this->assertEquals($expected, $queue[0]);
+    }
+
+    public function testAddPrimaryKey()
+    {
+        $tableName = 'users';
+        $actual    = $this->object->useTable($tableName);
+        $this->assertTrue($actual);
+
+        $columnName = 'uid';
+
+        $this->object->addPrimaryKey($tableName, $columnName);
+        $queue = $this->object->dumpQueue();
+        //var_dump($queue);
+        $expected = "ALTER TABLE `{$this->prefix}_{$tableName}` ADD PRIMARY KEY(`{$columnName}`)";
+        $this->assertEquals($expected, $queue[0]);
+    }
+
+    public function testAddTable()
+    {
+        // Remove the following lines when you implement this test.
+        $this->markTestIncomplete(
+            'This test has not been implemented yet.'
+        );
+    }
+
+    public function testUseTable()
+    {
+        $actual = $this->object->useTable('users');
+        $this->assertTrue($actual);
+
+        $actual = $this->object->useTable('system_nosuch_table');
+        $this->assertFalse($actual);
+    }
+
+    public function testGetColumnAttributes()
+    {
+        $tableName = 'users';
+        $actual    = $this->object->useTable($tableName);
+        $this->assertTrue($actual);
+
+        $columnName = 'uid';
+
+        $actual = $this->object->getColumnAttributes($tableName, $columnName);
+
+        $this->assertNotFalse(stristr($actual, 'mediumint(8)'));
```

Comment:
I'm converting ("downporting") these tests from XOOPS 2.6.0: https://github.com/XOOPS/XoopsCore/tree/master/tests/unit, so most of the time I'm just focusing on making sure that they run, hoping that it will be a good start for all of us to review and improve them, while keeping them compatible to XOOPS 2.6.0 as much as we can, in order to make it easier to transition the XOOPS 2.5.11 code to XOOPS 2.6.0 by having similar tests.
Of course, the XOOPS 2.6.0 tests are pretty old, so some of the things there might be outdated, so if you see anything strange, please make a comment, and hopefully we'll be able to figure that out together, as we learn PHPUnit and Unit testing